quantum/process_keycode/process_swap_hands.c

```diff
@@ -0,0 +1,146 @@
+/* Copyright 2021
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "process_swap_hands.h"
+
+extern const keypos_t PROGMEM hand_swap_config[MATRIX_ROWS][MATRIX_COLS];
+
+static matrix_row_t swap_state[MATRIX_ROWS];
+
+static bool     swap_hands_enabled = false;
```

Comment:
Would be nice to have a callback to check and set this.

Reply:
callback as in:
```c
__attribute__((weak)) void swap_hands_changed(bool newState);
``

or api like:
```c
bool swap_hands_enabled();
void swap_hands_enable();
void swap_hands_disable();
void swap_hands_toggle();
```

Reply:
@drashna Any update?

Reply:
the second, specifically.

Reply:
cool, not really a "callback" but implementing now.